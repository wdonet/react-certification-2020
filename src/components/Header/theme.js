const base = {
  easeOutBack: 'cubic-bezier(0.34, 1.56, 0.64, 1)',
  colorWhite: 'rgb(255, 255, 255)',
  colorBlack: 'rgb(0, 0, 0)',
};

const dark = {
  id: 'dark',
  ...base,
  backgroundColor: '#333',
  textColor: 'black',
};

const light = {
  id: 'light',
  ...base,
  backgroundColor: '#666',
  textColor: 'white',
};

export const theme = { dark, light };
