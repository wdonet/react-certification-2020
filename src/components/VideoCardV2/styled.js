import styled from 'styled-components';

const VideoContainer = styled.div`
  display: flex;
  margin: 0 0 10px 0;
`;

const VideoImage = styled.img`
  float: left;
  width: 40%;
`;

const Title = styled.p`
  margin: 0;
  font-size: 0.8em;
  font-weight: 600;
  color: ${({ theme }) => theme.font.primary};
`;

const VideoDetails = styled.div`
  padding: 5px;
`;

const Details = styled.p`
  font-size: 0.8em;
`;

export { VideoContainer, VideoImage, Title, VideoDetails, Details };
