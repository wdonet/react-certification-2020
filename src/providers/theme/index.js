export { default, useTheme } from './Theme.provider';
