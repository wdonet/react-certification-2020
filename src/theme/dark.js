const dark = {
  palette: {
    primary: {
      main: '#556cd6',
      transparent: '3,45,70,0.4',
    },
    secondary: {
      main: '#303030',
      light: '#424242',
    },
    accent: '#19857b',
  },
  typography: {
    title: {
      color: '#fff',
    },
    text: {
      color: '#fafafa',
    },
  },
  appBar: {
    height: {
      desktop: '64px',
    },
  },
  profile: {
    primary: '#303030',
    secondary: '#636363',
  },
};

export default dark;
