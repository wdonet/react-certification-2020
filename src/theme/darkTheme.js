const darkTheme = {
  background: {
    primary: '#C5C5C5',
    secondary: '#333333',
    tertiary: '#202020',
    highlight: '#eea2a2',
  },
  font: {
    primary: '#FFFFFF',
    secondary: '#EFEFEF',
    tertiary: '#C5C5C5',
    highlight: '#eea2a2',
  },
};

export default darkTheme;
