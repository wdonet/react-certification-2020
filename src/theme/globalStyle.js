import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
html {
  font-size: 1.125rem;
  line-height: 1.6;
  font-weight: 400;
  font-family: sans-serif;
  box-sizing: border-box;
  scroll-behavior: smooth;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
}

*,
*::before,
*::after {
  box-sizing: inherit;
}

body {
  margin: 0;
  padding: 0;
  text-rendering: optimizeLegibility;
  background-color: ${({ theme }) => theme.background.tertiary};
  background-position: var(--bg-position);
  transition: background-position 2s ease;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.separator::before {
  content: 'â€¢';
  color: white;
  padding: 0.4rem;
}

`;

export default GlobalStyle;
