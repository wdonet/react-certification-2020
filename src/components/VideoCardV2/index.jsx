import React from 'react';

import { VideoContainer, VideoImage, Title, VideoDetails, Details } from './styled';

const VideosList = ({ image, title, details }) => {
  return (
    <VideoContainer>
      <VideoImage src={image} />
      <VideoDetails>
        <Title>{title}</Title>
        <Details>{details}</Details>
      </VideoDetails>
    </VideoContainer>
  );
};

export default VideosList;
