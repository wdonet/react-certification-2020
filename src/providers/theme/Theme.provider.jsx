// React
import React, { useState, useEffect, useContext, useCallback } from 'react';
// Utils
import { THEME_SELECTED } from '../../utils/constants';
import { storage } from '../../utils/storage';
// Themes
import light from '../../theme/light';
import dark from '../../theme/dark';

const ThemeContext = React.createContext({ theme: light, selected: false });

const useTheme = () => {
  const context = useContext(ThemeContext);
  if (!context) {
    throw new Error(`Can't use "useTheme" without an ThemeProvider!`);
  }
  return context;
};

const ThemeBaseProvider = ({ children }) => {
  const [theme, setTheme] = useState(light);
  const [selected, setSelected] = useState(false);

  useEffect(() => {
    const lastTheme = storage.get(THEME_SELECTED);
    const whichTheme = lastTheme ? dark : light;
    setSelected(lastTheme);
    setTheme(whichTheme);
  }, []);

  const toggleTheme = useCallback((toggle) => {
    setTheme(toggle ? dark : light);
    storage.set(THEME_SELECTED, toggle);
  }, []);

  return (
    <ThemeContext.Provider value={{ theme, selected, toggleTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

export { useTheme, ThemeContext };
export default ThemeBaseProvider;
