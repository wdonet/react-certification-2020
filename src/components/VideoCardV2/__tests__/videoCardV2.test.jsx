import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';

import VideoCardV2 from '../index';

import { lightTheme } from '../../../theme';

describe('VideoCardV2', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={lightTheme}>
          <VideoCardV2
            image="urlimage.com"
            title="title of the video"
            channelTitle="wizeline"
          />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
