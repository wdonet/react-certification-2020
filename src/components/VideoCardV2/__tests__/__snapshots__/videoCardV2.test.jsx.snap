// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCardV2 renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 0 10px 0;
}

.c1 {
  float: left;
  width: 40%;
}

.c3 {
  margin: 0;
  font-size: 0.8em;
  font-weight: 600;
  color: #333333;
}

.c2 {
  padding: 5px;
}

.c4 {
  font-size: 0.8em;
}

<div
  className="c0"
>
  <img
    className="c1"
    src="urlimage.com"
  />
  <div
    className="c2"
  >
    <p
      className="c3"
    >
      title of the video
    </p>
    <p
      className="c4"
    />
  </div>
</div>
`;
